Automattic/Gravatar-SDK-iOS
# Image Squaring Stategy

Closes #364 

### Description

After a discussion, we have decided that this will be an internal implementation, and not configurable by users of the SDK.

This updates the `AvatarService` `upload(:::)` function to automatically square images that are nearly square.

### Testing Steps
Use the following demo image:
506 x 500 (squarable image)
<img width="253" alt="506x500" src="https://github.com/user-attachments/assets/00747434-d130-4391-8bb7-a79e94d22dab">

550 x 500 (non-squarable image)
<img width="275" alt="550x500" src="https://github.com/user-attachments/assets/be53c5b9-5ea1-4937-be61-cb083c83ad32">

In the Demo app, bypass the custom image squaring that it does:
https://github.com/Automattic/Gravatar-SDK-iOS/blob/b6fd4689160737c3a137fa72b4f0d294652e51d7/Sources/GravatarUI/ImageCropper/ImageCropperViewController.swift#L185

Return the `croppedUIImage` instead of the squared version:
```diff
let croppedUIImage = UIImage(cgImage: croppedCGImage, scale: UITraitCollection.current.displayScale, orientation: .up)

guard let result = croppedUIImage.square(maxLength: Constants.maxOutputImageSizeInPixels) else { return }
- onCompletion?(result)
+ onCompletion?(croppedUIImage)
```

#### Test squarable image is squared
1. Launch the Demo app
2. Select the "Upload Image" demo
3. Enter the appropriate email/token
4. Select the squarable image (506 x 500)
5. Upload
   - [ ] OVSERVE that the upload is successful
6. Check your Gravatar profile on the web:
   - [ ] OBSERVE that the image was cropped (the 1px-wide red bars on left/right have been cropped out)

#### Test non-squarable image fails
1. Launch the Demo app
2. Select the "Upload Image" demo
3. Enter the appropriate email/token
4. Select the non-squarable image (550 x 500)
5. Upload
   - [ ] OVSERVE that the upload returns an error
6. Check your Gravatar profile on the web:
   - [ ] OBSERVE that the image was not uploaded
